test/functional/tests/ci/test_recovery.py

```diff
@@ -0,0 +1,227 @@
+#
+# Copyright(c) 2019-2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+#
+from os.path import basename, realpath
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache_config import CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskTypeLowerThan, DiskTypeSet, DiskType, Disk
+from test_tools.dd import Dd
+from test_tools.disk_utils import Filesystem, unmount, mount
+from test_tools.fs_utils import check_if_file_exists
+from test_utils.filesystem.file import File
+from test_utils.os_utils import sync
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+
+mount_point = "/mnt/cas"
+
+
+@pytest.mark.CI
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_recover_cache():
+    """
+        title: Recovery after turning off/on devices
+        description: |
+            Test data integrity after turning off cache device.
+        pass_criteria:
+          - Cache devices successfully loaded with metadata after turning devices off/on
+          - md5sums before and after all operations match each other
+          - creation, mount, unmount of filesystems succeeds
+    """
+    with TestRun.step("Prepare devices."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)] * 3)
+        cache_devs = cache_disk.partitions
+        core_disk = TestRun.disks["core"]
+        core_disk.create_partitions([Size(4, Unit.GibiByte)] * 3)
+        core_devs = core_disk.partitions
+
+    with TestRun.step("Start caches and add cores."):
+        caches, cores = [], []
+        for (cache_dev, core_dev) in zip(cache_devs, core_devs):
+            cache = casadm.start_cache(cache_dev, cache_mode=CacheMode.WB)
+            core = cache.add_core(core_dev)
+            caches.append(cache)
+            cores.append(core)
+
+    with TestRun.step("Create filesystem on core devices."):
+        filesystems = [Filesystem.xfs, Filesystem.ext3, Filesystem.ext4]
+        for (core, filesystem) in zip(cores, filesystems):
+            core.create_filesystem(filesystem)
+
+    with TestRun.step("Mount cache devices."):
+        core_mount_points = []
```

Comment:
I don't think you need to maintain core_mount_points, this is just an equivalent of [c.mount_point for c in cores[